### PR TITLE
Move Condition stuff to apis, add a v1beta1 Status.

### PR DESCRIPTION
**Pull request by mattmoor**
This moves the common Condition stuff to apis, and creates a v1beta1 form of Status that uses the Condition it defines (changing this in v1alpha1 is too breaking).

There aren't really any meaningful changes in this PR, mostly reorganization.  Enumerating what I did:
1. Copied `condition_set*.go` to `apis/`,
1. Copied the `Condition` portions of `conditions_types.go` to `apis/`,
1. Copied the balance of `conditions_types.go` to `apis/duck/v1beta1/status_types.go`,
1. Changed the parts of the above to reference things in the appropriate new places,
1. Removed the reflection-based `ConditionsAccessor` stuff, implementing it instead on `duckv1beta1.Status`.
1. Incorporate: https://github.com/knative/pkg/pull/358

cc @vdemeester (since you commented on the predecessors)

FYI I was able to import this into serving without any code changes, then lots to take advantage of it.

### CODE REVIEW

File: apis/duck/v1beta1/status_types_test.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	}, {
+		name: "Message",
+		raw:  `{"message":"this is just a test"}`,
+		condition: &apis.Condition{
+			Message: "this is just a test",
+		},
+	}, {
+		name: "all",
+		raw: `{
+				"type":"Foo",
```
Comment (mattmoor-sockpuppet) [inline apis/duck/v1beta1/status_types_test.go]:
```suggestion
				"type":"Foo",
```

File: apis/duck/v1beta1/status_types_test.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+		name: "Message",
+		raw:  `{"message":"this is just a test"}`,
+		condition: &apis.Condition{
+			Message: "this is just a test",
+		},
+	}, {
+		name: "all",
+		raw: `{
+				"type":"Foo", 
+				"status":"True",
```
Comment (mattmoor-sockpuppet) [inline apis/duck/v1beta1/status_types_test.go]:
```suggestion
				"status":"True",
```

**Comment by mattmoor** [comment]
/hold 

Putting a hold on this, but I'd like to land it once we cut 0.5

**Comment by mattmoor** [comment]
You can see my WIP import into serving [here](https://github.com/knative/serving/compare/master...mattmoor:v1beta1-status?expand=1), although I'll probably split it across a few PRs.

**Comment by mattmoor** [comment]
I'll fix coverage too :(

**Comment by mattmoor** [comment]
This needs: https://github.com/knative/pkg/pull/358

**Comment by mattmoor** [comment]
/hold cancel

This creates a second parallel implementation, so it won't disrupt anyone if it goes in early.

**Pull request merged**